src/main/java/uk/gov/hmcts/divorce/document/DocumentGenerator.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.divorce.document;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.type.Document;
+import uk.gov.hmcts.divorce.divorcecase.model.Applicant;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.document.content.templatecontent.TemplateContent;
+import uk.gov.hmcts.divorce.document.model.DocumentType;
+import uk.gov.hmcts.divorce.document.print.documentpack.DocumentPackInfo;
+import uk.gov.hmcts.divorce.document.print.model.Letter;
+
+import java.time.Clock;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.time.LocalDateTime.now;
+import static org.springframework.util.CollectionUtils.firstElement;
+import static uk.gov.hmcts.divorce.caseworker.service.task.util.FileNameUtil.formatDocumentName;
+import static uk.gov.hmcts.divorce.document.DocumentUtil.getLettersBasedOnContactPrivacy;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class DocumentGenerator {
+
+    private final List<TemplateContent> templateContent;
```

Comment:
should this be called `allTemplates`?

Reply:
Well they're not templates though, they handle adding template content. But agreed, bad name - have changed to allTemplateContentHandlers